src/LuaUtils.cpp

```diff
@@ -171,127 +171,363 @@ static bool _import_core(lua_State *L, const std::string &importName)
 
 #undef DEBUG_IMPORT
 #ifdef DEBUG_IMPORT
-#define DEBUG_PRINTF Output
+	#define DEBUG_PRINTF Output
+	#define DEBUG_INDENTED_PRINTF IndentedOutput
+	#define DEBUG_INDENT_INCREASE IndentIncrease
+	#define DEBUG_INDENT_DECREASE IndentDecrease
 #else
-#define DEBUG_PRINTF(...)
+	#define DEBUG_PRINTF(...)
+	#define DEBUG_INDENTED_PRINTF(...)
+	#define DEBUG_INDENT_INCREASE(...)
+	#define DEBUG_INDENT_DECREASE(...)
 #endif
 
-static bool _import(lua_State *L, const std::string &importName)
+// simple struct to pass data between functions
+struct ImportInfo
 {
+	const std::string& importName; // original name argument from "import(importName)"
+	std::string& fileName; // name of the existing file
+	const bool& isFullName; // if true import will no try to load relative to the importDirectories
+	std::vector<std::string>& importDirectories; // contans list of paths where file can be imported
+};
+
+/**
+* Returns path of last element in the lua stack
+*/
+static std::string get_caller(lua_State *L)
+{
+	assert(L);
 	LUA_DEBUG_START(L);
 
-	DEBUG_PRINTF("import: request for %s\n", importName.c_str());
+	std::string caller;
+	lua_Debug ar;
 
-	// get the cache of previously imported things
-	lua_getfield(L, LUA_REGISTRYINDEX, "Imports");
+	// if have stack and caller in the stack
+	if (lua_getstack(L, 1, &ar) && lua_getinfo(L, "S", &ar) && ar.source)
+	{
+		int start = 0;
+		int end = strlen(ar.source);
+
+		if (ar.source[start] != '@' && end > 5)
```

Comment:
Please add a comment regarding the nature of the prefix you're removing here?
Also, I'd be more comfortable with braces around the start=4; block.

Reply:
@laarmen sorry but it isnt my block of code, I just moved it from old import to separate function. Dont know for sure how it works. If need I can to understand the code and change somehow; please tell.

Reply:
Fair enough. Could you please drop a comment with a TODO or XXX marker on it? Like XXX: Extraneous copy ?
TIA.